src/librustc_typeck/check/method/probe.rs

```diff
@@ -307,9 +321,13 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                 // a real method lookup, this is a hard error (it's an
                 // ambiguity and we can't make progress).
                 if !is_suggestion.0 {
-                    let t = self.structurally_resolved_type(span, final_ty);
-                    assert_eq!(t, self.tcx.types.err);
-                    return None
+                    // don't report a type error if we dereferenced a raw pointer,
```

Comment:
Is that really a problem in practice?

Reply:
It broke something in libstd: https://github.com/rust-lang/rust/blob/707d0708158dabd9b0a12af21a4b51bf1713cb9e/src/libstd/sys_common/gnu/libbacktrace.rs#L76

Here's a minimal reproduction:

```
fn main() {
    let data = std::ptr::null();
    let _ = &data as *const *const ();
    if data.is_null() {}
}
```

Reply:
I get that. I think we should probably make this a future-compat warning, or an error with the `arbitrary_self_types` feature flag set.